### ISSUE TITLE
Page unresponsive on 10K Socket Connections

### ISSUE THREAD

**Issue by vishwassingh47** [socketio/socket.io-admin-ui#22]
Page is not responsive when when large number of sockets are connected (~10K).

Have tested in 2 scenarios:
1. Start SocketIO server,Open Admin UI Page, now add load 10K socket connections in realtime, 10K connections are instantly connected on the server but Admin UI Page couldn't keep up with the real count.It then became unresponsive as it is with flooded with realtime docket data.

2. Start SocketIO server,add load 10K socket connections,now open Admin UI,it will take long to connect and then page became unresponsive. 

![Screenshot from 2021-07-07 00-05-10](https://user-images.githubusercontent.com/30126596/124651295-5b2da980-deb8-11eb-9b27-316069eca260.png)

**Comment by darrachequesne** [comment]
Yes, that's a lot of updates for the UI, which makes it unsuitable for more than a thousand connections.

We could add a lazy mode, where specific sockets/rooms are not tracked. WDYT?

**Comment by vishwassingh47** [comment]
Yes instead of broadcasting each and every events, we could make it lazy loaded or support batch operations.
That way it can handle Very Large Number of connections.

**Comment by md-seb** [comment]
> Yes, that's a lot of updates for the UI, which makes it unsuitable for more than a thousand connections.
> 
> We could add a lazy mode, where specific sockets/rooms are not tracked. WDYT?

I think the server should not push, but client should pull instead, whenever the UI thinks it's ready. Updating every 2 secs is perhaps too fast anyway - items are jumping around too fast on screen when there's a lot of activity. Maybe 5 seconds would be better as a minimum?

Here's my idea for lazy mode. I call it the Adaptive Refresh.
Configure minRefresh speed to e.g. 5000ms and refreshRatio to e.g. 1.2
UI sends a refresh request to the server and measures the roundtrip
UI refreshes screen
UI creates setTimeout for the next refresh to be Math.min(roundtrip*refreshRatio, minRefresh)

This would self-adapt to server load and also response size with just a few (two or so) requests, depending on websocket activity.

**Comment by hatemragab** [comment]
Are any updates about this issue?

**Comment by darrachequesne** [comment]
For future readers:

You can now use the production model, which was implemented in https://github.com/socketio/socket.io-admin-ui/commit/e0d91cadb11205c5f2c686c239a50cb2eef9795d and included in version `0.4.0`.

In this mode, the server send fewer data to the admin UI, so it is suitable for a bigger workload.

Please reopen if needed!
